### PR TITLE
Feature/test page

### PR DESCRIPTION
**Pull request by hmayenve**
# Name of Feature 
`feature/test-page`
### Related Tickets

- [QTC-166](https://jira.amida.com/browse/QTC-166)

### How Things Worked (or Didn't) Before This PR
There want's a page for running DBQ tests on dbq

### How Things Work Now
A new page is create to upload a mapping and run dbq_tests on it.

## How to Test
- `npm start`
- `localhost:1776/test`

## Readiness

1. [ ] This PR has full test coverage (If this PR fixes a bug, you _must_ add test cases respresentative of the bug).
2. [ ] This PR passes all automated tests.
3. [ ] This PR has no linting errors.
4. [ ] This PR has no hardcoded UI strings or other obvious issues.
5. [ ] In addition to testing my changes while running this code via `npm start`, I tested my changes while running this code on my machine via `npm run build-prod-start` (necessary because the latter command mimics production).
6. [ ] This PR's changes to configuration files have been documented in all appropriate places (such as but not limited to README.md), if applicable.
7. [ ] This PR's required changes outside of the scope of this repository have been documented in this pull request.

### CODE REVIEW

File: src/components/FileUpload.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import config from 'config/configFromCookie'
+import 'styles/qtcstyles.scss'
+
+const { applicationPresentationName } = config
+const LOCAL_API_URL = `http://127.0.0.1:5000/test_dbq`
```
Comment (puckybreg) [inline src/components/FileUpload.jsx]: Let's set both of these in the `config` and request them from there

File: src/components/FileUpload.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import config from 'config/configFromCookie'
+import 'styles/qtcstyles.scss'
```
Comment (puckybreg) [inline src/components/FileUpload.jsx]: An update from develop will change this but I believe it is now called just `styles`
```suggestion
import 'styles/styles.scss'
```

File: src/components/FileUpload.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+class FileUpload extends Component {
+    constructor(props) {
+        super(props);
+    
+        this.state = {
+          dbqnum: '',
+          uploadInput: ''
+        };
+    
+        this.handleUpload = this.handleUpload.bind(this);
```
Comment (puckybreg) [inline src/components/FileUpload.jsx]: See @mountHouli  comments about binding class functions https://github.com/amida-tech/qtc-web-ui/pull/4#discussion_r299716547

File: src/components/FileUpload.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          method: 'POST',
+          body: data,
+        }).then(response => {
+            window.location.assign(response.url);
+        }).catch(err => console.log("error in api", err))
+      }
+    
+      render() {
+        return (
+            <div className='app-page' id='test-page-div'>
```
Comment (puckybreg) [inline src/components/FileUpload.jsx]: See @mountHouli comments about unused ids https://github.com/amida-tech/qtc-web-ui/pull/4#discussion_r299716003

File: src/components/FileUpload.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                <h2>Testing a Mapping</h2>
+                Please enter a DBQ number corresponding to the mapping file you would like to test.
```
Comment (puckybreg) [inline src/components/FileUpload.jsx]: I think you should put this text in a `<p/>` div

**Review by puckybreg — CHANGES_REQUESTED** [review]
Left a few small comments. I think what might be the best process is for you to update from develop and then read through @mountHouli's comments on Elizabeth's Navigation PR because I think many of the comments will be identical.

**Comment by hmayenve** [comment]
Made updates

File: src/components/TestPage.jsx [hunk 6]
```diff
@@ -1,20 +1,82 @@
...
+      dbqnum: '',
+      uploadInput: null,
+      results: '',
+      isTesting: false,
+    };
+
+    this.handleUpload = this.handleUpload.bind(this);
+    this.baseState = this.state
+  }
+  resetForm = () => {
```
Comment (puckybreg) [inline src/components/TestPage.jsx:25]: This works great!

File: src/components/TestPage.jsx [hunk 7]
```diff
@@ -1,20 +1,82 @@
...
+    this.setState({ isTesting: true })
+    const data = new FormData();
+    data.append('file', this.uploadInput.files[0]);
+    data.append('dbqnum', this.dbqnum.value);
+    fetch(apiURL+'test_dbq', {
+      method: 'POST',
+      body: data,
+    }).then(res => res.text())
+      .then(text => this.setState({ results: text, isTesting: false}), (error) => { 
+        if (error){
```
Comment (puckybreg) [inline src/components/TestPage.jsx:41]: Just a note, no action required, but if you don't have the API running, The error that you get "Cannot POST" doesn't get caught by this conditional

**Review by puckybreg — APPROVED** [review]
Great Work @hmayenve 🏅

**Pull request merged**